exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -948,6 +981,26 @@ interface ISqlExpressionBuilder {
     infix fun List<Column<*>>.notInList(list: Iterable<List<*>>): InListOrNotInListBaseOp<List<*>> =
         MultipleInListOp(this, list, isInList = false)
 
+    /**
+     * Checks if all columns in this `List` are not equal to any of the [CompositeID]s from [list].
+     *
+     * @sample org.jetbrains.exposed.sql.tests.shared.entities.CompositeIdTableEntityTest.testInListWithCompositeIdEntities
+     **/
+    @Suppress("UNCHECKED_CAST")
+    @JvmName("notInListCompositeIDs")
+    @LowPriorityInOverloadResolution
+    infix fun List<Column<*>>.notInList(list: Iterable<CompositeID>): InListOrNotInListBaseOp<List<*>> {
+        val componentList = list.map { id ->
+            List(this.size) { i ->
+                val component = id[this[i] as Column<Comparable<Any>>]
+                component.takeIf {
+                    it !is EntityID<*> || this[i].columnType is EntityIDColumnType<*>
```

Comment:
Minor: I expect that it's just extra check, but is it possible that non-`EntityId` column (actually here is `Comparable`, but I expect that it's always columns) has `EntityIDColumnType`, does it make sense?

By the way removing of `it !is EntityID<*> ||` does not break tests (at least fo H2 and Postgres)

Reply:
You're right. The tests were failing before (for case with double-wrapped EntityID<EntityID>) but not anymore, so I must have changed something in the logic lower down (hopefully it's not the tests that changed).